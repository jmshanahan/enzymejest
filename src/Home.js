import React, { Component } from 'react';

import './App.css';

const Test = () => <div>Testing</div>;

function fetchBoxes() {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve(['First Box', 'Second Box', 'Third Box']);
    }, 1000);
  });
}

export default class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      on: false,
      input: '',
      mainColor: 'blue',
      boxes: [],
      lifeCycle: ''
    };
  }
  componentDidMount() {
    this.setState({ lifeCycle: 'componentDidMountut' });

    fetchBoxes().then(boxes => this.setState({ boxes }));
  }
  componentWillReceiveProps() {
    this.setState({ lifeCycle: 'componentWillReceiveProps' });
  }
  handleStrings(str) {
    if (str === 'Hello World') return true;
    else false;
  }

  render() {
    return (
      <div className="App">
        <h1 className="App-title">Welcome to React</h1>
        <h3 className={this.state.mainColor}>Everyone is Welcome</h3>
        <p className="lifeCycle">{this.state.lifeCycle}</p>
        <p className="button-state">{this.state.on ? 'Yes!' : 'No!'}</p>
        <button onClick={() => this.setState({ on: true })}>Click</button>
        <h2>{this.state.input}</h2>
        <input
          onChange={e => this.setState({ input: e.currentTarget.value })}
          type="text"
        />
      </div>
    );
  }
}
