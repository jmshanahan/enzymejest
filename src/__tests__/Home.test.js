import React from 'react';
import '../../enzyme.config';
import Home from '../Home';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('App shallow rendering', () => {
  it('should render app', () => {
    const wrapper = shallow(<Home />, {
      context: {},
      disableLifecycleMethods: true
    });
    expect(
      wrapper
        .find('h1')
        .first()
        .text()
    ).toBe('Welcome to React');
    // console.log(wrapper.debug());
  });
  it('matches the snapshot', () => {
    const wrapper = shallow(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('on button click changes p text', () => {
    const wrapper = shallow(<Home />);
    const button = wrapper.find('button');
    expect(wrapper.find('.button-state').text()).toBe('No!');
    button.simulate('click');
    expect(wrapper.find('.button-state').text()).toBe('Yes!');
  });
  it('on input change, title changes text', () => {
    const wrapper = shallow(<Home />);
    const input = wrapper.find('input');
    expect(wrapper.find('h2').text()).toBe('');
    input.simulate('change', { currentTarget: { value: 'Tyler' } });
    expect(wrapper.find('h2').text()).toBe('Tyler');
  });
  it('updates className on new state', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.find('.blue').length).toBe(1);
    wrapper.setState({ mainColor: 'red' });
    expect(wrapper.find('.blue').length).toBe(0);
    expect(wrapper.find('.red').length).toBe(1);
  });
  it('calls componentDidMount', () => {
    jest.spyOn(Home.prototype, 'componentDidMount');
    const wrapper = shallow(<Home />);
    expect(Home.prototype.componentDidMount.mock.calls.length).toBe(1);
    expect(wrapper.find('.lifeCycle').text()).toMatch(/componentDidMount/);
  });
  it('setProps calls componentWillReceiveProps', () => {
    jest.spyOn(Home.prototype, 'componentWillReceiveProps');
    const wrapper = shallow(<Home />);
    wrapper.setProps({ hide: true });
    expect(Home.prototype.componentWillReceiveProps.mock.calls.length).toBe(1);
    expect(Home.prototype.componentWillReceiveProps.mock.calls.length).toBe(1);
  });
  it('handleStrings function returns correctly', () => {
    const wrapper = shallow(<Home />);
    const trueReturn = wrapper.instance().handleStrings('Hello World');
    const falseReturn = wrapper.instance().handleStrings('');
    expect(trueReturn).toBeTruthy();
    expect(falseReturn).toBeFalsy();
  });
});
describe('App mount rendering ', () => {
  it('should render app', () => {
    const wrapper = mount(<Home />);
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
    wrapper.unmount();
  });
  it('matches the snapshot', () => {
    const wrapper = mount(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
