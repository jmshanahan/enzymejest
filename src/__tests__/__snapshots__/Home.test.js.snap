// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App mount rendering  matches the snapshot 1`] = `
<Home>
  <div
    className="App"
  >
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Everyone is Welcome
    </h3>
    <p
      className="lifeCycle"
    >
      componentDidMountut
    </p>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
</Home>
`;

exports[`App shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="App-title"
  >
    Welcome to React
  </h1>
  <h3
    className="blue"
  >
    Everyone is Welcome
  </h3>
  <p
    className="lifeCycle"
  >
    componentDidMountut
  </p>
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
